common/internal_model/src/main/scala/uk/ac/wellcome/models/Implicits.scala

```diff
@@ -0,0 +1,161 @@
+package uk.ac.wellcome.models
+
+import io.circe.generic.semiauto._
+import io.circe._
+
+import scala.collection.immutable.::
+
+import uk.ac.wellcome.models.work.internal.{
+  TransformedBaseWork,
+  IdentifiedBaseWork,
+  UnidentifiedWork,
+  UnidentifiedInvisibleWork,
+  UnidentifiedRedirectedWork,
+  IdentifiedWork,
+  BaseWork,
+  InstantRange,
+  Period,
+  ProductionEvent,
+  Person,
+  Contributor,
+  Identifiable,
+  Unidentifiable,
+  Item,
+  Genre,
+  DigitalLocation,
+  PhysicalLocation,
+  Location,
+  Concept,
+  AbstractRootConcept,
+  MaybeDisplayable,
+  AbstractAgent,
+  AbstractConcept,
+  SourceIdentifier,
+  MergeCandidate,
+  WorkType,
+  Displayable,
+  Identified,
+  Subject,
+}
+import uk.ac.wellcome.models.matcher.MatcherResult
+import uk.ac.wellcome.json.JsonUtil._
+
+object Implicits {
```

Comment:
This file is pretty ugly :see_no_evil: 

One option to make it more readable and maintainable would be to write a macro to generate the implicits below. Not sure if worth the effort right now though :thinking: